File: web/app/types/ai.ts
/** AI provider domain types (Phase 9). Raw tokens are never present here. */

import type { ISODateString } from './api'

export type AiProviderStatus = 'NOT_CONFIGURED' | 'UNTESTED' | 'CONNECTED' | 'ERROR' | 'DISABLED'

export interface AiProvider {
  id: string
  provider: string
  model: string
  base_url: string | null
  configured: boolean
  api_key_masked: string | null
  enabled: boolean
  is_default: boolean
  status: AiProviderStatus
  last_tested_at: ISODateString | null
  last_error: string | null
  created_at: ISODateString
  updated_at: ISODateString
}

export interface AiProviderCreateInput {
  provider: string
  model: string
  api_key: string
  base_url?: string
  enabled?: boolean
  make_default?: boolean
}

export interface AiProviderUpdateInput {
  model?: string
  api_key?: string
  base_url?: string
  enabled?: boolean
}

export interface AiProviderModel {
  id: string
  label: string | null
}

export interface AiProviderTestResult {
  ok: boolean
  status: string
  detail: string | null
  models: AiProviderModel[]
}

export interface AiProviderCatalogItem {
  key: string
  requires_base_url: boolean
  default_base_url: string | null
}

export interface AiProviderCatalog {
  items: AiProviderCatalogItem[]
}

File: backend/app/schemas/ai.py
"""AI provider API schemas. Raw credentials are never part of a response."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import ProviderStatus


class ProviderCreateRequest(BaseModel):
    provider: str = Field(min_length=1, max_length=32)
    model: str = Field(min_length=1, max_length=128)
    api_key: str = Field(min_length=1, max_length=512)
    base_url: str | None = Field(default=None, max_length=255)
    enabled: bool = True
    make_default: bool = False


class ProviderUpdateRequest(BaseModel):
    model: str | None = Field(default=None, max_length=128)
    api_key: str | None = Field(default=None, max_length=512)
    base_url: str | None = Field(default=None, max_length=255)
    enabled: bool | None = None


class ProviderSchema(BaseModel):
    id: uuid.UUID
    provider: str
    model: str
    base_url: str | None
    configured: bool
    api_key_masked: str | None
    enabled: bool
    is_default: bool
    status: ProviderStatus
    last_tested_at: datetime | None
    last_error: str | None
    created_at: datetime
    updated_at: datetime


class ProviderModelSchema(BaseModel):
    id: str
    label: str | None = None


class ProviderTestResponse(BaseModel):
    ok: bool
    status: str
    detail: str | None = None
    models: list[ProviderModelSchema] = Field(default_factory=list)


class ProviderCatalogItem(BaseModel):
    key: str
    requires_base_url: bool
    default_base_url: str | None = None


class ProviderCatalogSchema(BaseModel):
    items: list[ProviderCatalogItem]
